pkg/build/nodeimage/internal/kube/builder_docker.go

```diff
@@ -83,29 +90,35 @@ func (b *dockerBuilder) Build() (Bits, error) {
 		},
 		os.Environ()...,
 	)
-	// build binaries
+	// binaries we want to build
 	what := []string{
 		// binaries we use directly
 		"cmd/kubeadm",
 		"cmd/kubectl",
 		"cmd/kubelet",
 	}
-	cmd := exec.Command(
-		"build/run.sh",
-		"make", "all", "WHAT="+strings.Join(what, " "),
-	).SetEnv(env...)
-	exec.InheritOutput(cmd)
-	if err := cmd.Run(); err != nil {
-		return nil, errors.Wrap(err, "failed to build binaries")
-	}
 
-	// build images
-	cmd = exec.Command("make", "quick-release-images").SetEnv(env...)
+	// build images + binaries (binaries only on 1.21+)
+	cmd := exec.Command("make", "quick-release-images", "KUBE_EXTRA_WHAT="+strings.Join(what, " ")).SetEnv(env...)
```

Comment:
why added here and not in env?

Reply:
because it's not common across commands and make supports it either way.